src/components/TodosFilter.tsx

```diff
@@ -0,0 +1,80 @@
+import { useContext } from 'react';
+import { TodosContext } from './TodosContext';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+export const TodosFilter: React.FC = () => {
+  const {
+    todos,
+    handleStatus,
+    selectedStatus,
+    setTodos,
+  } = useContext(TodosContext);
+
+  const ItemsLeft = () => {
+    return todos.filter(todo => todo.completed === false).length;
+  };
+
+  const handleDeleteClick = () => {
+    const filterTodos = (prevTodos: Todo[]) => prevTodos
+      .filter((todoitem) => todoitem.completed === false);
+
+    setTodos(filterTodos(todos));
+  };
+
+  return (
+    <>
+      {todos.length > 0 && (
+        <footer className="footer">
+          <span className="todo-count" data-cy="todosCounter">
+            {`${ItemsLeft()} items left`}
```

Comment:
You are calling `ItemsLeft()` function twice. It's better to call it once and store the result in a variable, then use that variable. This will improve performance.